src/Confluent.Kafka/ConfigPropertyNames.cs

```diff
@@ -0,0 +1,198 @@
+
+
+        // ---- Client
+
+        /// <summary>
+        ///     Name of the configuration property that specifies a delegate for
+        ///     handling log messages. If not specified, a default callback that
+        ///     writes to stderr will be used.
+        /// </summary>
+        /// <remarks>
+        ///     By default not many log messages are generated.
+        ///
+        ///     For more verbose logging, specify one or more debug contexts 
+        ///     using the 'debug' configuration property. The 'log_level'
+        ///     configuration property is also relevant, however logging is
+        ///     verbose by default given a debug context has been specified,
+        ///     so you typically shouldn't adjust this value.
+        ///
+        ///     Warning: Log handlers are called spontaneously from internal 
+        ///     librdkafka threads and the application must not call any 
+        ///     Confluent.Kafka APIs from within a log handler or perform any
+        ///     prolonged operations.
+        /// </remarks>
+        public const string LogDelegateName = "log.delegate";
+
+
+        // ---- Deserializers
+
+        /// <summary>
+        ///     Name of the configuration parameter used to specify the encoding
+        ///     when deserializing string keys.
+        /// </summary>
+        public const string DeserializerKeyEncodingConfigParam = "dotnet.string.deserializer.encoding.key";
+
+        /// <summary>
+        ///     Name of the configuration parameter used to specify the encoding
+        ///     when deserializing string values.
+        /// </summary>
+        public const string DeserializerValueEncodingConfigParam = "dotnet.string.deserializer.encoding.value";
+
+
+        // ---- Serializers
+
+        /// <summary>
+        ///     Name of the configuration parameter used to specify the encoding 
+        ///     when serializing string keys.
+        /// </summary>
+        public const string SerializerKeyEncodingConfigParam = "dotnet.string.serializer.encoding.key";
+
+        /// <summary>
+        ///     Name of the configuration parameter used to specify the encoding 
+        ///     when serializing string values.
+        /// </summary>
+        public const string SerializerValueEncodingConfigParam = "dotnet.string.serializer.encoding.value";
+
+
+        /// <summary>
+        ///     Name of the configuration parameter used to specify the initial size 
+        ///     (in bytes) of the buffer used for Avro message serialization. Use a value 
+        ///     high enough to avoid resizing the buffer, but small enough to avoid 
+        ///     excessive memory use. Inspect the size of the byte array returned by the
+        ///     Serialize method to estimate an appropriate value. Note: each call to 
+        ///     serialize creates a new buffer.
+        /// 
+        ///     default: 128
```

Comment:
two good points. I'll add the metric as a todo and increase the default now.

Reply:
related: i'm thinking about how to change the serializer api more generally - shift the burden of memory allocation / management to the producer rather than have the serializer do it. this would allow a memory pool that adapts to use automatically.